### ISSUE TITLE
Fix S3011 FN: Support UnsafeAccessorAttribute

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#8153]
While [S3011](https://sonarsource.github.io/rspec/#/rspec/S3011/csharp) explicitly talks about using reflection to get access to members that otherwise could not be accessed by direct dereference, the focus of the rule is on increased accessibility, and reflection is just a way to achieve that.

The [C# 12 Zero-overhead member access](https://github.com/dotnet/runtime/issues/81741) is a new way of achieving the same result, but without using reflection.

```cs
using System.Runtime.CompilerServices;

[UnsafeAccessor(UnsafeAccessorKind.Constructor)]
extern static UserData<T> CallPrivateConstructor<T>();

// This API allows accessing backing fields for auto-implemented properties with unspeakable names
[UnsafeAccessor(UnsafeAccessorKind.Field, Name = "<Name>k__BackingField")]
extern static ref string GetName<T>(UserData<T> obj);

UserData<string> ud = CallPrivateConstructor<string>();
ref string x = ref GetName(ud);
x = "Name";
Console.WriteLine(x);
Console.WriteLine(ud);

class UserData<T>
{
    private UserData() { }
    private T Name { get; }

    public override string ToString() => Name.ToString();
}
```

The rule should be extended to take into account this new feature. 
An alternative would to create a dedicated rule for `UnsafeAccessorAttribute`.

Related New Rule Idea: https://github.com/SonarSource/sonar-dotnet/issues/8151

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1668](https://sonarsource.atlassian.net/browse/NET-1668)

[NET-1668]: https://sonarsource.atlassian.net/browse/NET-1668?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
